doc/components/divider.md

```diff
@@ -0,0 +1,53 @@
+# Divider
+
+ROHD HCL provides an integer divider module to get the dividend of numerator and denominator operands. The divider implementation is not pipelined and has a maximum latency of the bit width of the operands.
+
+## Interface
+
+The inputs to the divider module are:
+
+* ```clock``` => clock for synchronous logic
+* ```reset``` => reset for synchronous logic (active high)
+* ```a``` => the numerator operand
+* ```b``` => the denominator operand
+* ```newInputs``` => indication that a new division operation is being requested
+
+The outputs of the divider module are:
+
+* ```c``` => the result of the division
+* ```divZero``` => divide by zero error indication
+* ```isReady``` => the result of the current division operation is ready
+* ```isBusy``` => the divider is currently busy working on a division operation
+
+The numerical inputs (```a```, ```b```, ```c```) are parametrized by a constructor parameter called ```dataWidth```. All other signals have a width of 1.
+
+## Protocol Description
+
+To initiate a new request, it is expected that the requestor drive ```newInputs``` to high along with the numerical values for ```a``` and ```b```. The first cycle in which ```isBusy``` is low where the above occurs is the cycle in which the operation is accepted by the divider.
```

Comment:
I think this is a ready/valid interface;   newInputs is validIn  and isBusy=Low is the readyOut back to the requestor.

validIn & readyOut is acceptance of the request.

His previous 'isReady' signal is validOut for the current result going downstream.

Reply:
Yup Dez's description is right. Though it does kind of bring up a good related question (port renaming aside) - should the output side have a valid/ready pair (whereas currently it only has a valid)?

Reply:
Perhaps that's a good idea -- the Divider could decide to hold the result on the output until it's "accepted" by `ready & valid` on the output side.  This would be a consistent pattern across ready/valid components (and make handshaking automation simpler in the future).